libcontainer/cgroups/fs2/misc.go

```diff
@@ -0,0 +1,52 @@
+package fs2
+
+import (
+	"bufio"
+	"os"
+	"strings"
+
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+	"github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
+)
+
+func statMisc(dirPath string, stats *cgroups.Stats) error {
+	for _, file := range []string{"current", "events"} {
+		fd, err := cgroups.OpenFile(dirPath, "misc."+file, os.O_RDONLY)
+		if err != nil {
+			return err
+		}
+
+		s := bufio.NewScanner(fd)
+		for s.Scan() {
+			key, value, err := fscommon.ParseKeyValue(s.Text())
+			if err != nil {
+				fd.Close()
+				return err
+			}
+
+			key = strings.TrimSuffix(key, ".max")
```

Comment:
I'm not familiar with `misc` controller, I don't know whether the kernel allow a resource name end with `.max` or not, for example: `res_a.max`, if yes, the key in `misc.events` will be `res_a.max.max`? So maybe we should not remove `.max` in `misc.current`.
And could you show us a url link which describes the format of `misc.events`?

Reply:
@lifubang the resource names are not arbitrary user provided keys but they are defined in `include/linux/misc_cgroup.h`. Currently

> And could you show us a url link which describes the format of misc.events?

The doc is https://docs.kernel.org/admin-guide/cgroup-v2.html#misc. It only defines `max` for each resource.

Reply:
> the resource names are not arbitrary user provided keys but they are defined in `include/linux/misc_cgroup.h`. Currently

https://github.com/torvalds/linux/blob/master/kernel/cgroup/misc.c#L19-L27
Yes, currently the value of resource name will be `sev` or `sev_es`.